impala/dbapi/__init__.py

```diff
@@ -27,29 +27,64 @@
 from impala.error import (Error, Warning, InterfaceError, DatabaseError,
                           InternalError, OperationalError, ProgrammingError,
                           IntegrityError, DataError, NotSupportedError)
-from impala.util import warn_deprecate_hs2
+from impala.util import warn_deprecate_hs2, warn_deprecate
+
+
+AUTH_MECHANISMS = ['NOSASL', 'PLAIN', 'GSSAPI', 'LDAP']
+
 
 # PEP 249 module globals
 apilevel = '2.0'
 threadsafety = 1  # Threads may share the module, but not connections
 paramstyle = 'pyformat'
 
+def connect(host='localhost', port=21050, protocol='hiveserver2', database=None,
+            use_ldap=None, ldap_user=None, ldap_password=None,  # deprecated
+            use_kerberos=None,  # deprecated
+            auth_mechanism=None, **kwargs):
```

Comment:
Let's not use `kwargs`.  Add the additional args explicitly.  I'd move the deprecated ones to the end.